core/src/main/java/io/undertow/io/AsyncReceiverImpl.java

```diff
@@ -498,15 +510,11 @@ public void receivePartialBytes(final PartialBytesCallback callback, final Error
         long contentLength;
         if (contentLengthString != null) {
             contentLength = Long.parseLong(contentLengthString);
-            if (contentLength > Integer.MAX_VALUE) {
-                error.error(exchange, new RequestToLargeException());
-                return;
-            }
         } else {
             contentLength = -1;
         }
-        if (maxBufferSize > 0) {
-            if (contentLength > maxBufferSize) {
+        if(maxContentSize > 0){
```

Comment:
Im not 100% keen on having another variable doing similar thing.

Reply:
Other than that it looks fine.

Reply:
Agreed.